crates/primitives/src/proof.rs

```diff
@@ -0,0 +1,38 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::{Deserialize, Serialize};
+
+use crate::buf::Buf32;
+
+#[derive(Clone, Debug, Copy, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum RollupVerifyingKey {
+    // Verifying Key used to verify proof created using SP1
+    SP1VerifyingKey(Buf32),
+    // Verifying Key used to verify proof created using Risc0
+    Risc0VerifyingKey(Buf32),
+}
+
+/// `ProofId` is an enumeration representing various identifiers for proofs.
+/// It is used to uniquely identify the proofing tasks for different types of proofs.
+#[derive(
+    Debug, Clone, Copy, PartialEq, BorshSerialize, BorshDeserialize, Serialize, Deserialize,
+)]
+pub enum ProofId {
```

Comment:
Maybe a better name would be `proof_key` since there's more semantic information associated with it now?